src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -0,0 +1,277 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * BlockTemplatesCompatibility class.
+ *
+ * To bridge the gap on compatibility with PHP hooks and blockified templates.
+ *
+ * @internal
+ */
+class BlockTemplatesCompatibility {
+
+	const LOOP_ITEM_ID = 'product-loop-item';
+
+	/**
+	 * Hoding the supported hooks, their positions and the hooked functions.
+	 *
+	 * @var array $hook_data
+	 */
+	protected $hook_data = array(
+		'core/query'            => array(
+			'woocommerce_before_main_content' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_output_content_wrapper' => 10,
+					'woocommerce_breadcrumb'             => 20,
+				),
+			),
+			'woocommerce_after_main_content'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_output_content_wrapper_end' => 10,
+				),
+			),
+		),
+		'core/post-title'       => array(
+			'woocommerce_before_shop_loop_item_title' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_show_product_loop_sale_flash' => 10,
+					'woocommerce_template_loop_product_thumbnail' => 10,
+				),
+			),
+			'woocommerce_shop_loop_item_title'        => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_template_loop_product_title' => 10,
+				),
+			),
+			'woocommerce_after_shop_loop_item_title'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_template_loop_rating' => 5,
+					'woocommerce_template_loop_price'  => 10,
+				),
+			),
+		),
+		self::LOOP_ITEM_ID      => array(
+			'woocommerce_before_shop_loop_item' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_template_loop_product_link_open' => 10,
+				),
+			),
+			'woocommerce_after_shop_loop_item'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_template_loop_product_link_close' => 5,
+					'woocommerce_template_loop_add_to_cart' => 10,
+				),
+			),
+		),
+		'core/post-template'    => array(
+			'woocommerce_before_shop_loop' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_output_all_notices' => 10,
+					'woocommerce_result_count'       => 20,
+					'woocommerce_catalog_ordering'   => 30,
+				),
+			),
+			'woocommerce_after_shop_loop'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_pagination' => 10,
+				),
+			),
+		),
+		'core/query-no-results' => array(
+			'woocommerce_no_products_found' => array(
```

Comment:
`woocommerce_no_products_found` seems to be triggered even if there are products in the loop:

![imatge](https://user-images.githubusercontent.com/3616980/212627164-c878ff26-4ed9-4daa-b027-2644106456bd.png)

Reply:
Yeah, I noticed the same issue.

Reply:
I fixed this issue by checking the block content before injecting the hooks.